Lib/test/eintrdata/eintr_tester.py

```diff
@@ -441,6 +444,16 @@ def test_select(self):
         self.stop_alarm()
         self.assertGreaterEqual(dt, self.sleep_time)
 
+    def test_select_long_signal(self):
+        rd, wr = os.pipe()
+        self.addCleanup(os.close, rd)
+        self.addCleanup(os.close, wr)
+        # timeout is enough for signal to fire, but not for it to return
+        timeout = self.signal_delay + self.signal_duration/2
+        rlist, _, _ = select.select([rd], [], [], timeout)
```

Comment:
Please use self.sleep_time as the timeout.

Reply:
But then the test doesn't necessarily trigger the bug!
As the comment states, we have to make sure the signal fires **before reaching the timeout**, but finishes running **after reaching it**. Any other case, no bug is triggered.

Reply:
Currently, a signal is sent every 100 ms, whereas the sleep is 200 ms. tearDown() ensures that the Python signal handler has been called at least once. See https://bugs.python.org/issue35316 ;-)

Reply:
Do I miss something? If timeout is 200ms:

1. Signal handler starts after some time, say 90ms -- it fires before reaching the timeout, which is okay
2. Signal handler finishes immediately (or within 50ms, because of the `time.sleep` I've added)
3. Less than 200ms have passed (140ms) -- *bug is not triggered*, thus the test case is basically useless!

Again, I need a very fine timing situation - handler starts before reaching the timeout, but finishes only after.